src/main/java/module/teamAnalyzer/ht/HattrickManager.java

```diff
@@ -1,54 +1,54 @@
 package module.teamAnalyzer.ht;
 
 import core.db.DBManager;
+import core.file.xml.TeamStats;
 import core.file.xml.XMLManager;
 import core.file.xml.XMLPlayersParser;
 import core.model.match.MatchKurzInfo;
 import core.net.MyConnector;
 import core.net.OnlineWorker;
 import core.util.HODateTime;
 import module.teamAnalyzer.manager.PlayerDataManager;
-import module.teamAnalyzer.vo.Filter;
-import module.teamAnalyzer.vo.Match;
-import module.teamAnalyzer.vo.PlayerInfo;
+import module.teamAnalyzer.vo.*;
 
 import java.time.temporal.ChronoUnit;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
 
-import module.teamAnalyzer.vo.SquadInfo;
 import org.w3c.dom.Document;
 
-
 /**
  * Hattrick Download Helper class
  *
  * @author <a href=mailto:draghetto@users.sourceforge.net>Massimiliano Amato</a>
  */
 public class HattrickManager {
 
+    private static final Map<Integer, Map<String, String>> teamDetailsCache = new HashMap<>();
+    private static final Map<Integer, Map<String, TeamStats>> seriesDetailsCache = new HashMap<>();
+
     /**
-     * Method that download from Hattrick the available matches for the team
-     * If manual filter, the last 30 is made available.
+     * Method that downloads from Hattrick the available matches for the team <code>teamId</code>.
+     * If manual filter, the last 30 are made available.
      * If auto filter, enough matches to supply the filter needs are available.
-     * Recent tournament are added if on manual, or if they are wanted, in addition to
+     *
+     * <p>Recent tournament are added if on manual, or if they are wanted, in addition to
      * the number specified.
      *
-     * @param teamId teamid to download matches for
+     * @param teamId ID of team to download matches for
      * @param filter the match filter object.
      */
     public static void downloadMatches(final int teamId, Filter filter) {
+        System.out.println("Current Thread: " + Thread.currentThread().getName()); // FIXME Remove printout
```

Comment:
@tychobrailleur when do you want to fix this?

Reply:
Fixed too